stake-pool/cli/src/main.rs

```diff
@@ -391,12 +391,80 @@ fn pool_tokens_to_stake_amount(pool_data: &StakePool, tokens: u64) -> u64 {
         .unwrap() as u64
 }
 
+struct WithdrawAccount {
+    pubkey: Pubkey,
+    account: Account,
+    amount: u64,
+}
+
+fn prepare_withdraw_accounts(
+    config: &Config,
+    pool_withdraw_authority: &Pubkey,
+    amount: u64,
+) -> Result<Vec<WithdrawAccount>, Error> {
+    let mut accounts = get_authority_accounts(config, &pool_withdraw_authority);
+    if accounts.is_empty() {
+        return Err("No accounts found.".to_string().into());
+    }
+    // Sort from lowest to highest balance
+    accounts.sort_by(|a, b| a.1.lamports.cmp(&b.1.lamports));
+
+    // Prepare the list of accounts to withdraw from
+    let mut withdraw_from: Vec<WithdrawAccount> = vec![];
+    let mut remaining_amount = amount;
+
+    // First use largest amounts
+    while !accounts.is_empty() && accounts.last().unwrap().1.lamports < remaining_amount {
```

Comment:
I'm finding the two loops a bit confusing, so maybe we could add some comments about what's being done. Here we iterate from the back to get large accounts, then we iterate from the front to get small accounts. What's the purpose of each one?  Perhaps we could get it all done in one iteration?

Reply:
Mostly because we already had the second loop and added first one to get the rest of the accounts. :). You're right it is better to make them into one loop.

Reply:
So now there is only one loop and it seems like there is more code now, but I think it looks prettier and with more comments now.
So the goal is to catch the following:

For accounts [10, 8, 7, 5, 3, 2]:
Withdraw 5 will return single claim of [5]
Withdraw 13 returns [10, 3]
Withdraw 14 returns [10, 4 of 5]
Withdraw 19 returns [10, 8, 1 of 2]

So it is not the most effective algorithm now, but should be enough.